src/confluent_kafka/src/Admin.c

```diff
@@ -1531,6 +1528,361 @@ const char Admin_list_consumer_groups_doc[] = PyDoc_STR(
+
+                } else if(PyObject_IsInstance(alteration,UserScramCredentialDeletion_type)){
+                        /* Deletion Type*/
+                        cfl_PyObject_GetAttr(alteration,"mechanism",&mechanism,NULL,0,0);
+                        if (!PyObject_IsInstance(mechanism, ScramMechanism_type)) {
+                                PyErr_SetString(PyExc_TypeError,
+                                        "mechanism should be of ScramMechanism type");
+                                goto err;
+                        }
+                        cfl_PyObject_GetInt(mechanism,"value",&mechanism_val,0,1);
+
+                        c_alterations[i] = rd_kafka_UserScramCredentialDeletion_new(user_c,(rd_kafka_ScramMechanism_t)mechanism_val);
+                        Py_DECREF(mechanism);
+                        mechanism = NULL;
+                }
+
+                Py_DECREF(u_user);
+                Py_XDECREF(uo_user);
+        }
+        /* Use librdkafka's background thread queue to automatically dispatch
+         * Admin_background_event_cb() when the admin operation is finished. */
+        rkqu = rd_kafka_queue_get_background(self->rk);
+
+        /*
+         * Call AlterConfigs
+         *
+         * We need to set up a CallState and release GIL here since
+         * the event_cb may be triggered immediately.
+         */
+        CallState_begin(self, &cs);
+        rd_kafka_AlterUserScramCredentials(self->rk, c_alterations, alteration_cnt, c_options, rkqu);
+        CallState_end(self, &cs);
+
+        if(c_alterations) {
+                for(i=0;i<alteration_cnt;i++)
+                        rd_kafka_UserScramCredentialAlteration_destroy(c_alterations[i]);
+                free(c_alterations);
+        }
+        rd_kafka_queue_destroy(rkqu); /* drop reference from get_background */
+        rd_kafka_AdminOptions_destroy(c_options);
+        Py_DECREF(UserScramCredentialAlteration_type); /* from lookup() */
+        Py_DECREF(UserScramCredentialUpsertion_type); /* from lookup() */
+        Py_DECREF(UserScramCredentialDeletion_type); /* from lookup() */
+        Py_DECREF(ScramCredentialInfo_type); /* from lookup() */
+        Py_DECREF(ScramMechanism_type); /* from lookup() */
+        Py_RETURN_NONE;
+err:
+
+        Py_XDECREF(u_user);
+        Py_XDECREF(uo_user);
+        Py_XDECREF(salt);
+        Py_XDECREF(password);
+        Py_XDECREF(scram_credential_info);
+        Py_XDECREF(mechanism);
+
+        Py_DECREF(UserScramCredentialAlteration_type); /* from lookup() */
+        Py_DECREF(UserScramCredentialUpsertion_type); /* from lookup() */
+        Py_DECREF(UserScramCredentialDeletion_type); /* from lookup() */
+        Py_DECREF(ScramCredentialInfo_type); /* from lookup() */
+        Py_DECREF(ScramMechanism_type); /* from lookup() */
+
+        if(c_alterations) {
+                for(i=0;i<alteration_cnt;i++)
+                        rd_kafka_UserScramCredentialAlteration_destroy(c_alterations[i]);
```

Comment:
1. Should be till `i` not till `alteration_cnt` as we can reach here in error case. In this case
2. Use this function instead - `rd_kafka_UserScramCredentialAlteration_destroy_array`

```suggestion
                rd_kafka_UserScramCredentialAlteration_destroy_array(c_alterations, i);
```